src/main/java/io/cdap/plugin/gcp/bigquery/relational/BigQueryRelation.java

```diff
@@ -507,6 +541,69 @@ protected static DeduplicateAggregationDefinition qualify(DeduplicateAggregation
     return builder.build();
   }
 
+  protected static boolean supportsWindowAggregationDefinition(WindowAggregationDefinition def) {
+    Collection<Expression> partitionExpressions =  def.getPartitionExpressions();
+    Collection<Expression> selectExpressions = def.getSelectExpressions().values();
+    Collection<WindowAggregationDefinition.OrderByExpression> orderByExpressions = def.getOrderByExpressions();
+    Collection<Expression> orderExpressions = new ArrayList<>(orderByExpressions.size());
+    for (WindowAggregationDefinition.OrderByExpression orderByExpression : orderByExpressions) {
+      orderExpressions.add(orderByExpression.getExpression());
+    }
+    return supportsExpressions(partitionExpressions) && supportsExpressions(selectExpressions)
+      && supportsExpressions(orderExpressions);
+  }
+  protected static WindowAggregationDefinition qualify(WindowAggregationDefinition def) {
+    WindowAggregationDefinition.Builder builder = WindowAggregationDefinition.builder();
+    builder.select(qualifyKeys(def.getSelectExpressions()));
```

Comment:
Why are we putting operations in separate statements rather than chaining operations like `builder.select(...).partition(...).orderBy` etc as is usually followed in a builder pattern?

Reply:
We can do either.Followed the pattern of other qualify() methods for group by and dedup